Feature: Refresh of remote control screen buttons

wutschel opened the pull request:
## Description
<!--- Detailed info for reviewers and developers -->
This PR is an add-on to https://github.com/xbmc/Official-Kodi-Remote-iOS/pull/311 and refreshes the icons/buttons of the remote control screen, as well as of the remote help screen. Screenshots can be found in #310.

## Summary for release notes
<!--- This will be shown in Testflight to end users / testers -->
<!--- Please fill in, as usually PR title isn't clear to ordinary users -->
<!--- If your changes don't modify app files, please add prefix [not app] -->
Feature: Refresh remote control screen buttons (aligned with Kodi 19)

wutschel commented:
I am adding this PR as work-in-progress to allow testing. It should be added only after the bigger code style and AppDelegate rework PRs to avoid problems rebasing those.

wutschel commented:
Rebased to master.

wutschel commented:
Rebased to master.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -54,22 +54,7 @@ -(void)hideButton:(NSArray *)buttonsToDo hide:(BOOL)hide{
-    }
-        
-    int newWidth = (int) (296 * transform);
+    CGFloat newWidth = remoteControlView.frame.size.width - ANCHORRIGHTPEEK;
```
please add underscores to the macro, it's quite difficult to read

wutschel replied:
You mean `ANCHORRIGHTPEEK`? If so, would you mind if I am doing this in a separate coding styles PR? This change touches several other files which are not connected to this PR.

kambala-decapitator replied:
sure that's ok

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -98,9 +81,15 @@ - (void)setEmbeddedView{
                     hide: YES];
     }
+    // Place the transitional view in the middle between the two button rows
+    CGFloat lowerButtonUpperBorder = CGRectGetMinY(((UIButton*)[self.view viewWithTag:21]).frame);
+    CGFloat upperButtonLowerBorder = CGRectGetMaxY(((UIButton*)[self.view viewWithTag:6]).frame);
```
no need to cast to `UIButton`

wutschel replied:
Ok, will be done with next uodate.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -128,9 +117,12 @@ - (void)setEmbeddedView{
+    frame.size.width = newWidth;
     subsInfoLabel.frame = frame;
+    
+    frame = self.view.superview.frame;
```
using superview's `frame` (as opposed to `bounds`) in subview's calculations is generally incorrect. What are you trying to achieve here?

wutschel replied:
I wanted to draw the background image ("remote_temp") with same relative dimensions for this use case (embedded view) as for the standard remote view. This background image is at the same hierarchy level as the remote control view itself. 
The embedded view is drawn into a cell of a height of only 50 (see call of `[remoteControllerView setEmbeddedView]` in `RightMenuViewController.m`). The same dimensions I get when using `self.view.superview.frame` inside `setEmbeddedView`.

wutschel replied:
I get the same desired behaviour if I remove

```
frame = self.view.superview.frame;
panFallbackImageView.frame = frame;
```

and set the frame after `setEmbeddedView` in `RightMenuViewController.m` like this:

```
[remoteControllerView setEmbeddedView];
remoteControllerView.panFallbackImageView.frame = cell.frame;
```

kambala-decapitator replied:
the latter looks much clearer, please use that

wutschel replied:
Done.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -143,14 +135,13 @@ - (void)configureView{
         frame.size.height = frame.size.height *transform;
         frame.size.width = frame.size.width*transform;
-        [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y, frame.size.width* SCALE_TO_REDUCE_BORDERS, frame.size.height* SCALE_TO_REDUCE_BORDERS)];
+        [remoteControlView setFrame:CGRectMake(frame.origin.x, frame.origin.y, frame.size.width, frame.size.height)];
```
```suggestion
        remoteControlView.frame = frame;
```

wutschel replied:
Ok, will be part of next update.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -161,8 +152,15 @@ - (void)configureView{
+        // Calculate the maximum possible scaling for the remote
+        CGFloat scaleFactorHorizontal = STACKSCROLL_WIDTH / remoteControlView.frame.size.width;
+        CGFloat minViewHeight = MIN(CGRectGetWidth(UIScreen.mainScreen.fixedCoordinateSpace.bounds), CGRectGetHeight(UIScreen.mainScreen.fixedCoordinateSpace.bounds)) - REMOTE_PADDING;
+        CGFloat scaleFactorVertical =  minViewHeight / remoteControlView.frame.size.height;
```
please use `CGRectGetHeight` for consistency

wutschel replied:
No problem, done with next update.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -161,8 +152,15 @@ - (void)configureView{
+        CGFloat scaleFactorVertical =  minViewHeight / remoteControlView.frame.size.height;
+        CGFloat transform = MIN(scaleFactorHorizontal, scaleFactorVertical);
+
+        CGFloat newWidth = remoteControlView.frame.size.width * transform;
+        CGFloat newHeight = remoteControlView.frame.size.height * transform;
```
`CGRectGet...`

wutschel replied:
No problem, done with next update.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -1258,17 +1258,20 @@ - (void)viewDidLoad{
         }
     }
     if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
+        CGRect frame = CGRectMake(self.view.bounds.size.width - 256, self.view.bounds.size.height - TOOLBAR_ICON_SIZE - TOOLBAR_SPACING, TOOLBAR_ICON_SIZE, TOOLBAR_ICON_SIZE);
```
what is 256?

wutschel replied:
Argh, I already defined `TOOLBAR_START` for this. It also has the correct value, 256 was even a bit off. Will be part of next update.

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -1258,17 +1258,20 @@ - (void)viewDidLoad{
         
         UIButton *gestureButton = [UIButton buttonWithType:UIButtonTypeCustom];
-        gestureButton.frame = CGRectMake(self.view.bounds.size.width - 188, self.view.bounds.size.height - 43, 56, 36);
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
FYI it's simpler to use `+=` operator:
```suggestion
        frame.origin.x += TOOLBAR_PADDING;
```

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -1278,24 +1281,25 @@ - (void)viewDidLoad{
         UIButton *keyboardButton = [UIButton buttonWithType:UIButtonTypeCustom];
-        keyboardButton.frame = CGRectMake(self.view.bounds.size.width - 120, self.view.bounds.size.height - 43, 56, 36);
-        UIImage* keyboardImg = [UIImage imageNamed:@"keyboard_icon"];
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
+=

kambala-decapitator commented on XBMC Remote/RemoteController.m:
```diff
@@ -1278,24 +1281,25 @@ - (void)viewDidLoad{
 
-        UIButton *helpButton = [UIButton buttonWithType:UIButtonTypeInfoLight];
+        UIButton *helpButton = [UIButton buttonWithType:UIButtonTypeCustom];
+        frame.origin.x = frame.origin.x + TOOLBAR_PADDING;
```
+=

wutschel replied:
Ok, will do this the other 3 occasions as well.

kambala-decapitator reviewed (changes requested):
> Only keep remote images with 2x resolution

are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.

wutschel commented:
> are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.

I was wondering if we really need the x1 images as there are also no 1x images for the gesture zone. Of course I can easily add the 1x images for remote help. Should I?

kambala-decapitator commented:
> > are you sure that's correct? We support iOS 9 which iPad 2 is capable of running, and it's non-retina.
> 
> I was wondering if we really need the x1 images as there are also no 1x images for the gesture zone. Of course I can easily add the 1x images for remote help. Should I?

yes please

wutschel commented:
Rebased to master and updated regarding the review comments, except 

- `ANCHORRIGHTPEEK`: Will be done in separate PR
- Using superview's `frame` in `setEmbeddedView `: Still open

wutschel commented:
Resolved last review comment regarding `setEmbeddedView`.
